### PR TITLE
INF-950 - Switching docker actions to use main by default

### PR DESCRIPTION
**Pull request by scallister**
Some CICD processes are failing because of `master` and `main` being used in different places. This standardizes on `main` going forward, but on any error fails back to the prior behavior of `master`.

### CODE REVIEW

**Review by jmatt — APPROVED** [review]
I think this will work. The logic is sound but not familiar with the library.

:shipit:

File: docker/docker.go [hunk 1]
```diff
@@ -147,8 +148,38 @@ func (d *Docker) Tag() string {
...
 
-func (d *Docker) latestBranch() string {
-	return d.Env.GetDefault("DOCKER_LATEST_BRANCH", "master")
+// latestBranch - Returns branch that should be used for DOCKER_LATEST_BRANCH
+// If DOCKER_LATEST_BRANCH is not defined, returns "main"
+// If branch "main" does not exist in repository, returns "master"
+// On any error, returns "master" to maintain original compatibility
+func (d *Docker) latestBranch() (latestBranch string) {
+	// Leaving "master" as default to maintain compatibility if checking git repo fails
+	latestBranch = d.Env.GetDefault("DOCKER_LATEST_BRANCH", "master")
```
Comment (cep21) [inline docker/docker.go:157]: If someone explicitly sets DOCKER_LATEST_BRANCH, you should not do the main/master logic and return what they set.

**Review by cep21 — COMMENTED** [review]
I think it breaks if they set DOCKER_LATEST_BRANCH

File: docker/docker.go (hunk 1 above)
Comment (scallister) [reply docker/docker.go:157]: Good catch. Here is a PR that will make `DOCKER_LATEST_BRANCH` be better respected:
https://github.com/cresta/magehelper/pull/12

**Pull request merged**